e2e/utils.ts

```diff
@@ -7,3 +7,12 @@ export function transformStringToSlug(string: string) {
 export function getCollectionEndpoint() {
   return COLLECTIONS_ENDPOINT + transformStringToSlug(`${COLLECTION_NAME}`);
 }
+
+export function getRandomValueFromArray(array: string[]) {
```

Comment:
I believe we should avoid randomness and if there turn out to be multiple values that need to be tested add all of them to the tests rather than occasionally stumbling upon them making the issue harder to identify.

Reply:
Hm, there are a few reasons why I've added this: 1) I think we don't care about the country or state we choose in the test; 2) If I make country and state constants, the test will select and assert the same values every time, but the main purpose of the test is to verify if they can be changed successfully.
So I think we should test different combinations here.

Reply:
In this case rather than having them be random I'd increment through a list of values. This also avoids possible issues if the same random value is selected twice in a row.

Reply:
Okay, I understand, then I'd suggest checking the current selected value of the Country field to ensure we won't pick it up next. So in this case, we'll select another country each time, and then just select any option of the State.

Reply:
This addresses the issue where you don't want to select the same option twice in a row but that was really an aside, my original concern is that having the tests use randomness at all will mean that if a subset of the values cause an issue the tests will become flaky instead of failing every time. I'd like to strictly follow the principle that tests should be the same each time they run. One approach that uses incrementing values rather than randomly selecting could look like this:

```
  async selectNextOption(locator: Locator, currentOption: string) {
    const allOptions = await locator.getByRole('option').allTextContents();
    const currentIndex = allOptions.indexOf(currentOption);
    const nextOption = currentIndex + 1 === allOptions.length ? 0 : allOptions[currentIndex + 1]

    await locator.selectOption(nextOption);
    await this.page.waitForTimeout(1000); // wait for selection done

    return option;
  }
```

Reply:
I'm curious how this will work for the State. Now it looks like we will always select the second option, because after we select the Country, the State field will be reset to the first option, isn't it?

Reply:
Anyway, I'm ready to push the changes, but is this really what we need?

Reply:
The `selectNextOption` idea was intended to be a useful general utility but I see it isn't specifically useful for this PR so I am open to other options to remove the random element. Here are some alternative suggestions:

One approach would be to select fixed values:

```ts
  async selectCountry(country: string) {
    await this.shippingCountrySelect().selectOption(country);
    await this.verifyInputValue({ locator: this.shippingCountrySelect(), text: country });
    await this.page.waitForLoadState('domcontentloaded'); // wait for State options to load
  }

  async selectState(state: string) {
    this.shippingStateSelect().selectOption(state);
    await this.verifyInputValue({ locator: this.shippingStateSelect(), text: state });
  }
```

```ts
const country = 'Canada';
const state = 'Ontario';

...

await test.step('Update selection fields', async () => {
  await accountPage.selectCountry(country);
  await accountPage.selectState(state);
});
```

Another approach would be repeatable randomness: the values could be selected randomly using a seed so that the same "random" values are used on each execution. JavaScript's built-in `Math.random` does not support this but we could include an NPM package like `random-seed`. The only thing to consider there is if the execution order is not predictable the generator could not be shared across tests whose order is not predictable.

```ts
export function getRandomValueFromArray(rand: any, array: string[]) {
  if (!array.length) {
    throw new Error(`Array should not be empty.`);
  }

  const randomIndex = Math.floor(rand(array.length));
  return array[randomIndex];
}
```

```ts
import gen from 'random-seed';

...

await test.step('Update selection fields', async () => {
  const rand = gen.create('my seed value');
  country = await accountPage.selectRandomCountry(rand);
  state = await accountPage.selectRandomState(rand);
});
```

Reply:
done